internal/controller/controllers/bmh_agent_controller_test.go

```diff
@@ -426,4 +435,115 @@ var _ = Describe("bmac reconcile", func() {
+							"1001:db8::10/120",
+						},
+						MacAddress: macStr,
+					},
+				},
+				Disks: []v1beta1.HostDisk{
+					{Path: "/dev/sda", Bootable: true},
+					{Path: "/dev/sdb", Bootable: false},
+				},
+			}
+			clusterName := "test-cluster"
+			pullSecretName := "pull-secret"
+
+			agent.Spec.Role = models.HostRoleWorker
+			agent.Spec.ClusterDeploymentName = &v1beta1.ClusterReference{Name: clusterName, Namespace: testNamespace}
+			Expect(c.Create(ctx, agent)).To(BeNil())
+
+			image := &bmh_v1alpha1.Image{URL: "http://buzz.lightyear.io/discovery-image.iso"}
+			host = newBMH("bmh-reconcile", &bmh_v1alpha1.BareMetalHostSpec{Image: image, BootMACAddress: macStr, BMC: bmh_v1alpha1.BMCDetails{CredentialsName: fmt.Sprintf(adminKubeConfigStringTemplate, clusterName)}})
+			Expect(c.Create(ctx, host)).To(BeNil())
+
+			defaultClusterSpec := getDefaultClusterDeploymentSpec(clusterName, pullSecretName)
+			cluster = newClusterDeployment(clusterName, testNamespace, defaultClusterSpec)
+			cluster.Spec.Installed = true
+			Expect(c.Create(ctx, cluster)).ShouldNot(HaveOccurred())
+
+			secretName := fmt.Sprintf(adminKubeConfigStringTemplate, clusterName)
+			secret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretName,
+					Namespace: cluster.Namespace,
+				},
+				Data: map[string][]byte{
+					"kubeconfig": []byte(BASIC_KUBECONFIG),
+				},
+			}
+			Expect(c.Create(ctx, secret)).ShouldNot(HaveOccurred())
+
+			spokeMachine := &machinev1beta1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-cluster-bmh-reconcile",
+					Namespace: "test-namespace",
+					Labels: map[string]string{
+						machinev1beta1.MachineClusterIDLabel: clusterName,
+						MACHINE_ROLE:                         string(models.HostRoleWorker),
+						MACHINE_TYPE:                         string(models.HostRoleWorker),
+					},
+				},
+			}
+			Expect(c.Create(ctx, spokeMachine)).To(BeNil())
+		})
+
+		AfterEach(func() {
+			Expect(c.Delete(ctx, host)).ShouldNot(HaveOccurred())
+			Expect(c.Delete(ctx, agent)).ShouldNot(HaveOccurred())
+			Expect(c.Delete(ctx, cluster)).ShouldNot(HaveOccurred())
+		})
+
+		Context("when agent role worker and cluster deployment is set", func() {
+			It("should set spoke BMH", func() {
+				result, err := bmhr.Reconcile(ctx, newBMHRequest(host))
+				Expect(err).To(BeNil())
+				Expect(result).To(Equal(ctrl.Result{}))
+
+				updatedHost := &bmh_v1alpha1.BareMetalHost{}
+				err = c.Get(ctx, types.NamespacedName{Name: host.Name, Namespace: testNamespace}, updatedHost)
+				Expect(err).To(BeNil())
+				Expect(updatedHost.ObjectMeta.Annotations).To(HaveKey(BMH_HARDWARE_DETAILS_ANNOTATION))
+
+				spokeBMH := &bmh_v1alpha1.BareMetalHost{}
+				spokeClient := bmhr.spokeClient
+				err = spokeClient.Get(ctx, types.NamespacedName{Name: host.Name, Namespace: testNamespace}, spokeBMH)
+				Expect(err).To(BeNil())
+				Expect(spokeBMH.ObjectMeta.Annotations).To(HaveKey(BMH_HARDWARE_DETAILS_ANNOTATION))
```

Comment:
I would suggest going one step further and also assert that the contents are also the same.

`Expect(spokeBMH.ObjectMeta.Annotations[BMH_HARDWARE_DETAILS_ANNOTATION]).To(Equal(updatedHost.ObjectMeta.Annotations[BMH_HARDWARE_DETAILS_ANNOTATION]))
				`

Reply:
Done